### DISCUSSION TITLE
zh/config/global

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#56] [Announcements]
# zh/config/global

AList v3的文档

https://alist.nn.ci/zh/config/global.html

**Comment by Ethan930717** [comment]
Docker版本是否无法添加背景图片？


<style>
.chakra-ui-light{
  background-image: url("http://dahu.asia:7990/storage/backgrounds/zDf89jdZuSyxMUvIRq39wo7FHDULABmjjfOeqIPn.jpg") !important;
  background-repeat:no-repeat;background-size:cover;background-attachment:fixed;background-position-x:center;
}
.main-box {
  border-radius: 15px !important;
}
.chakra-ui-light .main-box {
  background-color: #ffffff70 !important;
}
.chakra-ui-light .readme-box {
  background-color: white !important;
}
.readme-box {
  border-radius: 15px !important;
}

同样代码在lxc容器（一键脚本）中生效，但是在docker版本中反复尝试都无法正常生效。已经排除了图片源的原因

**Comment by evlic** [comment]
> 只有加了密码的路径的直链才会有过期时间，否则不会过期。因为过期时间是加到 sign 查询参数中的，没有加密码的路径是不会检查 sign 的。

这一条可以顺带提一下全局开关，我的 markdown 引用同级目录的图片出现问题，为此我做了一些工作（debug 才看到有这个全局开关）给后来的朋友提个醒。

**Comment by kimycai** [comment]
将“签名所有”开启后，就没有被绕过密码访问私人文件的风险了吗？如果被绕过，有被删除文件的风险吗？

**Reply by zhendery** [reply]
看了一下这个签名的功能，只是用在复制文件链接的时候。
比如我有一个文件是http://xx.yy/abc/123.jpg， 如果不开签名，则我复制出来的文件直链就是http://xx.yy/abc/123.jpg， 把这个链接发给任**何没有登录的人**都可以直接下载这个 *123.jpg*。
而如果开启了签名，则复制出来的文件直链是http://xx.yy/abc/123.jpg?sign=ppssxxx， 即后面会跟上一串乱码（个人猜测是根据文件路径和当前时间hash出来的），只有把http://xx.yy/abc/123.jpg?sign=ppssxxx 这个链接发给**没有登陆的人**，他才可以下载*123.jpg*。而 **链接过期时间** 可以决定这个*ppssxxx*多久刷新一次，比如我设置了1（小时），即我发给别人的文件直链http://xx.yy/abc/123.jpg?sign=ppssxxx 在1小时就过期了，如果他没有及时下载那么需要我重新获取直链并分享给他。

因此该功能只对于文件分享有影响，不管开不开，持有直链的人都只能对你分享的这一个文件进行**读取**访问，而不可以进行删除。
而如果你是建了个用户给朋友使用，则开不开签名区别就不大了（除非你朋友也可能再给别人分享文件直链，因此我是建议开启的）。

另外我抓包看了下，在http下用户名密码是通过明文的方式发送到服务器的，虽然后续访问是用每次登录生成的随机token来降低被爆破的风险，但这个token也是能抓包抓到的，获取的文件内容更是全明文了。所以你应该担心的是这个，公网使用的话能https就尽量上s吧。

**Reply by zhendery** [reply]
上述只是我个人测试所得，不一定完全正确，仅供参考。
另外如果没有http又想公网使用，建议建立公网专用用户，关闭各种权限只保留读取权限，而管理员用户只在内网（家里局域网）登录，这样基本上就没啥风险了。

**Comment by youtehub** [comment]
ocr接口试试这个，可以直接使用，地址：
https://ocr-dwxg.onrender.com/ocr/file/json
填写到后台--设置--全局--OCR接口

**Reply by Winyer** [reply]
感谢，顺便备份下原OCR接口为：
http://api.nn.ci/ocr/file/json

**Comment by z4jst** [comment]
开启 webauthn ，个人资料里录入时提示： error validating origin
环境：NPM反代穿透至公网。公网HTTPS访问。
设备：iOS 17.1

**Comment by yeenjie123456** [comment]
我在打开webauthn时，提示The relying party ID is not a registrable domain suffix of, nor equal to the current domain.用的宝塔的nginx的反代，有https。

**Comment by bearrf** [comment]
打包下载 要怎么用？v3.36.0部署在服务器上部署的本机存储，在客户机访问后选择多文件，点打包下载弹提示框，点确定后提示 “当前状态: 下载文件中，不要关闭或刷新页面”就一直卡着没反应了。

**Comment by icexk** [comment]
设置了隐藏文件夹，在alist主页看不到，webdav挂载能看到隐藏的文件夹
